scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ClientHandshaker.java

```diff
@@ -223,11 +208,13 @@ protected void doProcessMessage(HandshakeMessage message) throws HandshakeExcept
 						String.format("Unsupported key exchange algorithm %s", getKeyExchangeAlgorithm().name()),
 						new AlertMessage(AlertLevel.FATAL, AlertDescription.HANDSHAKE_FAILURE, message.getPeer()));
 			}
+			handshakeMessages.add(message);
```

Comment:
~(clearly a details)
I feel this is a bit strange that sometime we call `handshakeMessages.add(message)` in the generic `doProcessMessage` and sometime in the specific processing method (like `receivedServerHello`).~

Fixed in next commit.